### ISSUE TITLE
JOSS Review

### ISSUE THREAD

**Issue by amoeba** [NESCent/TraitDB#222]
Hi @dleehr, I've had a chance to look over TraitDB and I appreciate the level of polish you've put on the documentation and the software as a whole. As the reviewer guidelines direct, I'm leaving comments specific to my review of your JOSS submission https://github.com/openjournals/joss-reviews/issues/1201 here.

I got a bit stuck at installation (configuring auth, really). Perhaps you could offer some assistance here. 

When I try to log in with Google, I get:

<img width="617" alt="screen shot 2019-02-07 at 9 01 52 pm" src="https://user-images.githubusercontent.com/563/52461464-a262b500-2b1b-11e9-8c0b-6085d59ec036.png">

I edited `devise.rb` and manually set the client ID and secret and login worked after that (I also had to set the callback URL in the Google Console). I'm running `rails server` in a shell with the two environmental variables set. I set my PostgreSQL environmental variables the same way and they were picked up fine. I'm using rbenv to manage rubies. What do you use?

**Other notes:**

- Your `Gemfile` specifies Ruby 2.2.0 but Bundler requires Ruby 2.3.0. I got an error from Bundler about this when running `bundle install` and worked around it by removing the Ruby version restriction from the `Gemfile`. Upgrading from Ruby 2.2.0 to 2.3.0 is quite a change for a JOSS review to request but, since others might encounter this, it might be a good idea to do at some point in the future.
- Get an OpenID link on login page links to "https://openid.net/get-an-openid/" which is a 404
- Without logging in, there is no Upload button. I think the text in the readme should be updated to reflect that.

- After logging in and upgrading my user to an admin, creating a new project worked. I then went to Upload, clicked Import Configs, clicked Upload New CSV Import Config, clicked Choose File, selected `lib/traitdb_import/Sample.yml`, clicked Upload CSV Import Config and got an error:

    >1 error prohibited this record from being saved
    >There were problems with the following fields:
    > 
    >Config file has contents that are not what they are reported to be

Any thoughts on my two issues here? I can take a look again tomorrow but have to run for now.

**Comment by dleehr** [comment]
Thanks for the feedback and for digging pretty deep to get past some holes in the documentation. I'll try to address those, but will be away next week. 

I too use rbenv and will try upgrading to 2.3.0. Will certainly address the issues. Thanks @amoeba !

**Comment by amoeba** [comment]
Regarding my issues with Devise earlier, I figured out that I was setting my env vars wrong. I use fish shel and I needed to add a `-x` to my `set` in order for the env vars to be shared with subprocesses. Installation work as documented.

Regarding my config upload issue, I tried again to get a config to upload and no matter which one I select I get:

> Config file has contents that are not what they are reported to be

I've tried the configs in `lib/traitdb_import` and also the example on the wiki with no luck. Am I doing something wrong here?

I ran through the other checklist items and have a couple of things. Here's the complete list of outstanding review items:

- **Version: Does the release version given match the GitHub release (v0.9.1)?**

  No, but this can be resolved once everything else is done.

- **Functionality: Have the functional claims of the software been confirmed?**

  Not yet, as I haven't gotten fully running due to not being able to import a config.

- **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

    Could you please add this to the readme?

* **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?


  - `ashman2014tree ` needs its DOI, https://doi.org/10.1038/sdata.2014.15


Everything else looks good. Could you please take a look at my issue with configs and let me know if I should be doing something differently? Thanks!

**Comment by dleehr** [comment]
Yep, will look at the configs today @amoeba. Thanks for your patience and diligence on the installation issues!, I was on a family vacation last week,

**Comment by dleehr** [comment]
Hey @amoeba I fixed the config loading issue in #225 and verified uploading `SampleDataset.csv` with `Sample.yml` config. However I discovered that the result display is broken (#226) so am working on that now.

**Comment by dleehr** [comment]
I've fixed the issues here, and released v1.0.0. I believe this is ready for another look now @amoeba

**Comment by amoeba** [comment]
Sounds great @dleehr. I can take another look before next week.

**Comment by amoeba** [comment]
Hey @dleehr, everything looks great now. I was able re-run initial setup, upload a config and matching data, and test out search. I've scanned the edits you've made to address my remaining issues and everything's satisfactory. I've updated the checklist over on https://github.com/openjournals/joss-reviews/issues/1201 and am about to walk over and ask the editor to finalize the submission.

Thanks for taking the time to work through my edits!
